### PR TITLE
Cluster webhook validation for update

### PR DESCRIPTION
**Pull request by mitalipaygude**
*Issue #, if available:*
https://github.com/aws/eks-anywhere/issues/1090

Description of changes:
Add cluster validations that we use in the CLI to the cluster webhook for updates.

Testing (if applicable):

-Unit tests
-Manual testing with tilt

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/api/v1alpha1/cluster_webhook.go [hunk 1]
```diff
@@ -94,16 +98,8 @@ func (r *Cluster) ValidateUpdate(old runtime.Object) error {
...
-		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "workerNodeGroupConfigurations"), r.Spec.WorkerNodeGroupConfigurations, err.Error()))
-	}
-
-	// Control plane configuration is mutable if workload cluster
-	if !r.IsSelfManaged() {
-		if err := validateControlPlaneLabels(r); err != nil {
-			allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "controlPlaneConfiguration", "labels"), r.Spec, err.Error()))
-		}
+	if err := r.Validate(); err != nil {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("Cluster"), r.Spec, err.Error()))
```
Comment (vivek-koppuru) [inline pkg/api/v1alpha1/cluster_webhook.go:102]: In a future pr, I'm assuming we will separate each error based on the specific field within `Validate` right?

File: pkg/api/v1alpha1/cluster_webhook.go [hunk 2]
```diff
@@ -86,6 +86,10 @@ func (r *Cluster) ValidateUpdate(old runtime.Object) error {
 		return apierrors.NewBadRequest(fmt.Sprintf("expected a Cluster but got a %T", old))
 	}
 
+	if r.IsSelfManaged() && !r.IsReconcilePaused() && features.IsActive(features.FullLifecycleAPI()) {
+		return apierrors.NewBadRequest(fmt.Sprintf("upgrading self managed clusters is not supported: %s", r.Name))
```
Comment (vivek-koppuru) [inline pkg/api/v1alpha1/cluster_webhook.go:90]: We need to update this for the `Create` function too right?

Comment (mitalipaygude) [reply pkg/api/v1alpha1/cluster_webhook.go:90]: yes, for create the logic will be updated to return the requests, and then validate the spec, similar to this logic. Will be doing it as a part of a separate PR though. This is for Update validations.

File: pkg/api/v1alpha1/cluster_webhook.go (hunk 1 above)
Comment (mitalipaygude) [reply pkg/api/v1alpha1/cluster_webhook.go:102]: yes, it will be a future refactor.

File: pkg/api/v1alpha1/cluster_webhook.go (hunk 2 above)
Comment (vivek-koppuru) [reply pkg/api/v1alpha1/cluster_webhook.go:90]: Also, what difference does it make checking for the envar in this case? We don't want to support it when the feature is enabled but when not, workload cluster is supported? Should it be separate conditions then?

**Review by vivek-koppuru — APPROVED** [review]

**Review by jiayiwang7 — APPROVED** [review]

File: pkg/api/v1alpha1/cluster_webhook.go (hunk 1 above)
Comment (g-gaston) [inline pkg/api/v1alpha1/cluster_webhook.go:102]: what about this?
```suggestion
		allErrs = append(allErrs, field.Invalid(field.NewPath("spec"), r.Spec, err.Error()))
```

**Review by jiayiwang7 — APPROVED** [review]

**Review by g-gaston — APPROVED** [review]

**Pull request merged**